### ISSUE TITLE
Error: object 'GT' not found when trying grur_imputations with a tidy .rad file

### ISSUE THREAD

**Issue by lehernandezc** [thierrygosselin/radiator#65]
Hello Thierry,

I am trying to impute missing genotypes from a VCF file with SNPs obtained in STACKS denovo pipeline using the grur_imputations() function. 
As adviced, I have used filter_rad() radiator function to tidy that VCF. The processes generated 13 folders, and the last folder contained what I think the tidy .rad file and filtered_strata information.

I used this .rad file and strata information to run the grur_imputations() function with method "rf_pred", however, I get the following error (see below), I have attached the required file to replicate the analysis, and a sample of my dataset in VCF format:

I have tried to delete VCF headers ##FORMAT not containing in my genotypes and only keeping,GT:DP:AD:GQ:GL, as you suggested in some issue reported previously but without success. I also set filter.monomorphic and filter.common.markers to FALSE because I though it was an issue with the filtering but I still had the same error object 'GT' not found.

I am not sure why this GT object is not created or if I am missing a crucial step.

I will appreciate your help, best wishes. Enrique

rhod.vcf <- grur::grur_imputations(
+   data = "./filter_rad_20190916@1424/13_filtered/radiator_data_20190916@1455.rad",
+   output = "vcf",
+   strata = "./filter_rad_20190916@1424/13_filtered/strata.filtered.tsv",
+   imputation.method = "rf_pred",
+   parallel.core = 1,
+   num.tree=5000)
################################################################################
################################################################################
Execution date/time: 20190916@1513
Imputation method: rf_pred
Hierarchical levels: strata

::grurgrur_imputations function call arguments:
    data = ./filter_rad_20190916@1424/13_filtered/radiator_data_20190916@1455.rad
    output = vcf
    strata = ./filter_rad_20190916@1424/13_filtered/strata.filtered.tsv
    imputation.method = rf_pred
    hierarchical.levels = strata
    verbose = TRUE
    parallel.core = 1
    filename = NULL

dots-dots-dots ... arguments

Arguments inside "..." assigned in ::grurgrur_imputations:
    num.tree = 5000

Default "..." arguments assigned in ::grurgrur_imputations:
    bagging_fraction = NULL
    bagging_freq = NULL
    boosting = NULL
    colsample_bytree = NULL
    cpu.boost = NULL
    early_stopping_rounds = NULL
    early_stopping_rounds = NULL
    eta = NULL
    feature_fraction = NULL
    gamma = NULL
    iteration.subsample = 1
    learning_rate = NULL
    max_depth = NULL
    min_child_weight = NULL
    min_data_in_leaf = NULL
    ncp = NULL
    nimpute = NULL
    nodesize = NULL
    nrounds = NULL
    nsplit = NULL
    num_leaves = NULL
    num_parallel_tree = NULL
    objective = NULL
    pmm = NULL
    random.seed = NULL
    save_name = NULL
    subsample = NULL
    subsample.markers = NULL

Random Forests options:
    number of trees: 5000
    minimum terminal node size: 1
    non-negative integer value used to specify random splitting: 10
    number of iterations: 10
    predictive mean matching: 0

Number of CPUs used during grur's operations: 1

Seed: 143316
Folder created: 15_radiator_genomic_converter_20190916@1513
File written: grur_imputations_args_20190916@1513.tsv
Analyzing strata file
    Number of strata: 25
    Number of individuals: 275
Synchronizing data and strata...
    Number of strata: 25
    Number of individuals: 275

Number of populations: 25
Number of individuals: 275
Number of markers: 1
Error: object 'GT' not found

Computation time, overall: 2 sec
############################## grur_imputations ################################

[grur_imputations_messages.txt](https://github.com/thierrygosselin/radiator/files/3616966/grur_imputations_messages.txt)
[Rsession.txt](https://github.com/thierrygosselin/radiator/files/3616967/Rsession.txt)

[grur_error_LH.zip](https://github.com/thierrygosselin/radiator/files/3616971/grur_error_LH.zip)

**Comment by thierrygosselin** [comment]
Hi lehernandezc,

it's a [known issue](https://github.com/thierrygosselin/grur/issues/10) I have yet come to fix and related to another package. I'll give it a try today to see if I can fix, otherwise the option/argument will be removed from grur.

Also, since it's a grur related issue, could you opened future bug in grur instead, thanks for reporting!

Thierry

**Comment by lehernandezc** [comment]
Hi Thierry,

I appreciate your response and time to look into this issue and I will keep an eye after you review. I attached previously a sample of my data, but I can provide the entire VCF if necessary.

I apologise, I will post future bug in the appropriate channels. 

Best wishes,

Enrique

**Comment by thierrygosselin** [comment]
Hello Enrique,

Update radiator and grur and try this with the data you provided:

```
library(radiator)
data <- radiator::read_vcf(data = "snps.loja.dplm.recode_example.vcf", strata = "popmap_grur.tsv")
ibm <- grur::missing_visualization(data = data)
```

It should work, if not, re-open the issue in grur.

Cheers
Thierry
